Objects/unicodeobject.c

```diff
@@ -427,6 +427,35 @@ get_error_handler_wide(const wchar_t *errors)
 }
 
 
+static inline int
+unicode_check_errors(const char *errors)
+{
+    PyInterpreterState *interp = _PyInterpreterState_GET_UNSAFE();
+#ifndef Py_DEBUG
+    /* Only enable check in development mode (-X dev). */
+    if (!interp->config.dev_mode) {
+        return 0;
+    }
+#endif
+    /* Avoid calling PyCodec_LookupError() before the codec registry is ready:
+       before_PyUnicode_InitEncodings() is called. */
+    if (!interp->fs_codec.encoding) {
+        return 0;
+    }
+
+    if (errors == NULL) {
```

Comment:
Would you move this to top of the function?
I think `errors == NULL` is the most common case.